sdk/python/feast/repo_config.py

```diff
@@ -1,80 +1,39 @@
-from enum import Enum
+import importlib
 from pathlib import Path
+from typing import Any
 
 import yaml
-from pydantic import (
-    BaseModel,
-    PositiveInt,
-    StrictInt,
-    StrictStr,
-    ValidationError,
-    root_validator,
-)
+from pydantic import BaseModel, StrictInt, StrictStr, ValidationError, root_validator
 from pydantic.error_wrappers import ErrorWrapper
 from pydantic.typing import Dict, Literal, Optional, Union
 
+from feast import errors
 from feast.telemetry import log_exceptions
 
+# This dict exists so that:
+# - existing values for the online store type in featurestore.yaml files continue to work in a backwards compatible way
+# - first party and third party implementations can use the same class loading code path.
+ONLINE_CONFIG_CLASS_FOR_TYPE = {
+    "sqlite": "feast.infra.online_stores.sqlite.SqliteOnlineStore",
+    "datastore": "feast.infra.online_stores.datastore.DatastoreOnlineStore",
+    "redis": "feast.infra.online_stores.redis.RedisOnlineStore",
+}
+
 
 class FeastBaseModel(BaseModel):
     """ Feast Pydantic Configuration Class """
 
     class Config:
         arbitrary_types_allowed = True
-        extra = "forbid"
-
-
-class SqliteOnlineStoreConfig(FeastBaseModel):
-    """ Online store config for local (SQLite-based) store """
-
-    type: Literal["sqlite"] = "sqlite"
-    """ Online store type selector"""
-
-    path: StrictStr = "data/online.db"
-    """ (optional) Path to sqlite db """
-
-
-class DatastoreOnlineStoreConfig(FeastBaseModel):
-    """ Online store config for GCP Datastore """
-
-    type: Literal["datastore"] = "datastore"
-    """ Online store type selector"""
-
-    project_id: Optional[StrictStr] = None
-    """ (optional) GCP Project Id """
-
-    namespace: Optional[StrictStr] = None
-    """ (optional) Datastore namespace """
-
-    write_concurrency: Optional[PositiveInt] = 40
-    """ (optional) Amount of threads to use when writing batches of feature rows into Datastore"""
-
-    write_batch_size: Optional[PositiveInt] = 50
-    """ (optional) Amount of feature rows per batch being written into Datastore"""
-
-
-class RedisType(str, Enum):
-    redis = "redis"
-    redis_cluster = "redis_cluster"
+        extra = "allow"
```

Comment:
Where are we using `extras`? Is this to support dynamic building of the whole config object?

Reply:
Yes, exactly. If we `forbid`, then anything that comes in from the store-specific Config object counts as "unexpected".